src/components/refactor/dialogs/generator/modification/generator-modification-form.js

```diff
@@ -0,0 +1,361 @@
+                currentNodeUuid,
+                'generators',
+                watchEquipmentId,
+                true
+            )
+                .then((value) => {
+                    if (value) {
+                        // when editing modification form, first render should not trigger this reset
+                        // which would empty the form instead of displaying data of existing form
+                        if (shouldEmptyFormOnGeneratorIdChangeRef?.current) {
+                            //creating empty table depending on existing generator
+                            let reactiveCapabilityCurvePoints = [];
+                            value?.reactiveCapabilityCurvePoints?.forEach(
+                                (element) => {
+                                    reactiveCapabilityCurvePoints.push({
+                                        [P]: null,
+                                        [Q_MIN_P]: null,
+                                        [Q_MAX_P]: null,
+                                        [OLD_P]: element.p,
+                                        [OLD_Q_MIN_P]: element.qminP,
+                                        [OLD_Q_MAX_P]: element.qmaxP,
+                                    });
+                                }
+                            );
+                            // resets all fields except EQUIPMENT_ID and REACTIVE_CAPABILITY_CURVE_TABLE
+                            resetForm(
+                                {
+                                    [EQUIPMENT_ID]: watchEquipmentId,
+                                    [REACTIVE_CAPABILITY_CURVE_TABLE]:
+                                        reactiveCapabilityCurvePoints,
+                                },
+                                true
+                            );
+                        }
+                        shouldEmptyFormOnGeneratorIdChangeRef.current = true;
+                        setGeneratorToModify(value);
+                    }
+                })
+                .catch(() => setGeneratorToModify(null));
+        } else {
+            resetForm();
+            setGeneratorToModify(null);
+        }
+    }, [
+        watchEquipmentId,
+        studyUuid,
+        currentNodeUuid,
+        setGeneratorToModify,
+        clearErrors,
+        resetForm,
+    ]);
+
+    const energySourceLabelId = getEnergySourceLabel(
+        generatorToModify?.energySource
+    );
+    const previousEnergySourceLabel = energySourceLabelId
+        ? intl.formatMessage({
+              id: energySourceLabelId,
+          })
+        : undefined;
+    const areIdsEqual = useCallback((val1, val2) => val1 === val2, []);
+
+    const generatorIdField = (
+        <AutocompleteInput
+            allowNewValue
+            name={EQUIPMENT_ID}
+            label={'ID'}
+            options={equipmentOptions}
+            formProps={{ ...filledTextField }}
+            size={'small'}
+            isOptionEqualToValue={areIdsEqual}
```

Comment:
add `freeSolo={false}` to add dropdown arrow

Reply:
allowNewValue in AutocompleteInput implies freeSolo true in Autocomplete and we have to set it to false to have dropdown arrow in field

Reply:
There is a prop to force dropdown arrow to appear